Sync from CLI without get permissions panics

Following was hit by a user when the token used to perform the sync did not have get permissions.
```
argocd app sync myapp --plaintext --timeout 60 --resource=:ConfigMap:my-configmap --resource=:Service:my-service --resource=apps:Deployment:my-deployment
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1426e0b]

goroutine 1 [running]:
github.com/argoproj/argo-cd/cmd/argocd/commands.printAppSummaryTable(0x0, 0xc000176880, 0x39)
    /go/src/github.com/argoproj/argo-cd/cmd/argocd/commands/app.go:216 +0x6b
github.com/argoproj/argo-cd/cmd/argocd/commands.waitOnApplicationStatus.func1(0x0)
    /go/src/github.com/argoproj/argo-cd/cmd/argocd/commands/app.go:1244 +0xe3
github.com/argoproj/argo-cd/cmd/argocd/commands.waitOnApplicationStatus(0x1a29b00, 0xc0000b7680, 0x7ffc31c6551f, 0xc, 0x3c, 0x10000, 0xc00013c0c0, 0x3, 0x4, 0x0, ...)
    /go/src/github.com/argoproj/argo-cd/cmd/argocd/commands/app.go:1301 +0x7fe
github.com/argoproj/argo-cd/cmd/argocd/commands.NewApplicationSyncCommand.func1(0xc000364900, 0xc0003558f0, 0x1, 0x7)
    /go/src/github.com/argoproj/argo-cd/cmd/argocd/commands/app.go:1093 +0x3f5
github.com/spf13/cobra.(*Command).execute(0xc000364900, 0xc000355880, 0x7, 0x7, 0xc000364900, 0xc000355880)
    /go/src/github.com/spf13/cobra/command.go:702 +0x2d3
github.com/spf13/cobra.(*Command).ExecuteC(0xc000524b40, 0xc000525b00, 0xc000524fc0, 0xc000354380)
    /go/src/github.com/spf13/cobra/command.go:783 +0x2dc
github.com/spf13/cobra.(*Command).Execute(0xc000524b40, 0xc000255f88, 0xc000096058)
    /go/src/github.com/spf13/cobra/command.go:736 +0x2b
main.main()
    /go/src/github.com/argoproj/argo-cd/cmd/argocd/main.go:14 +0x27
```

--- a/cmd/argocd/commands/app.go
+++ b/cmd/argocd/commands/app.go
@@ -1247,7 +1247,10 @@ func waitOnApplicationStatus(acdClient apiclient.Client, appName string, timeout
 
 	prevStates := make(map[string]*resourceState)
 	appEventCh := acdClient.WatchApplicationWithRetry(ctx, appName)
-	var app *argoappv1.Application
+	conn, appClient := acdClient.NewApplicationClientOrDie()
+	defer util.Close(conn)
+	app, err := appClient.Get(ctx, &application.ApplicationQuery{Name: &appName})
+	errors.CheckError(err)
 
 	for appEvent := range appEventCh {
 		app = &appEvent.Application

